frontend/src/components/NavBar.tsx

```diff
@@ -68,27 +23,34 @@ export const NavBar = () => {
   );
 
   return (
-    <NavbarDiv style={{ borderRadius: 50, backgroundColor: '#f5f5f5' }}>
-      <NavBarRightSide>
+    <div className={classes(css.navBarDiv)}>
+      <CornerPiece />
+      <div className={classes(css.navBarRightSide)}>
         {!userInfo && (
-          <NavbarLink as={Link} to={paths.loginPage}>
+          <Link className={classes(css.navBarLink)} to={paths.loginPage}>
             <Trans i18nKey="Login" />
-          </NavbarLink>
+          </Link>
         )}
         {userInfo && (
           <>
-            <NavbarLink as={Link} to={paths.logoutPage}>
-              <Trans i18nKey="Logout" />
-            </NavbarLink>
-            <NavbarLink as={Link} to={paths.userInfo}>
-              <Trans i18nKey="UserInfo" />
-            </NavbarLink>
+            <div className={classes(css.navBarText)}>
+              <Trans i18nKey="Viewing as" />
+            </div>
             <UserHotSwapWidget />
-            <NavbarDivider />
+            <div className={classes(css.navBarText)}>
+              <Trans i18nKey="Project" />
+            </div>
             <RoadmapSelectorWidget />
+            <div className={classes(css.navBarDivider)} />
+            <Link className={classes(css.navBarLink)} to={paths.userInfo}>
+              <PermIdentityIcon className={classes(css.icon)} />
+            </Link>
+            <Link className={classes(css.navBarLink)} to={paths.logoutPage}>
+              <PowerSettingsNewIcon className={classes(css.icon)} />
+            </Link>
```

Comment:
Näille linkeille olisi hyvä saada jokin hover tyyli, että käyttäjä saa jotakin visuaalista palautetta.
Myös klikattava alue on suhteellisesti melkoisesti suurempi korkeus suunnassa, ikonien ollessa jokseenkin neliöitä.

Nämä voisi designista kysäistä